.circleci/config.yml

```diff
@@ -1,88 +1,47 @@
-    image: carwow/elasticsearch-ci:7.6.1
-
-version: 2
 jobs:
-  bundle:
-    working_directory: ~/zelastic
-    docker:
-      - *ruby
-    steps:
-      - checkout
-      - restore_cache:
-          keys:
-            - bundle-{{ checksum "Gemfile.lock" }}
-            - bundle-
-      - run: |
-          bundle config --local path vendor/bundle &&
-          bundle check || bundle install --jobs=4 --retry=3
-          bundle clean --force
-      - save_cache:
-          key: bundle-{{ checksum "Gemfile.lock" }}
-          paths: [vendor/bundle]
-
-  rubocop:
-    working_directory: ~/zelastic
-    docker:
-      - *ruby
-    steps:
-      - checkout
-      - restore_cache: { keys: ['bundle-{{ checksum "Gemfile.lock" }}'] }
-      - run: bundle exec rubocop
-
-  tests:
-    working_directory: ~/zelastic
-    docker:
-      - *ruby
-      - *elasticsearch_container
-    steps:
-      - checkout
-      - restore_cache: { keys: ['bundle-{{ checksum "Gemfile.lock" }}'] }
-      - run:
-          name: Wait for ES to be ready
-          command: |
-            until curl $ELASTICSEARCH_URL/_cat/health | egrep '(green|yellow)' 2>&1 > /dev/null; do
-              echo -n .
-              sleep 1
-            done
-      - run: |
-          bundle exec rspec --pattern "**/*_spec.rb" --format "progress"
-
-  tests_7:
-    working_directory: ~/zelastic
+  test:
+    parameters:
+      ruby_version:
+        type: string
+      elasticsearch_version:
+        type: string
+      elasticsearch_gem_version:
+        type: string
     docker:
-      - *ruby
-      - *elasticsearch_7_container
+      - image: "cimg/ruby:<< parameters.ruby_version >>"
+      - image: "elasticsearch:<< parameters.elasticsearch_version >>"
+        environment:
+          "discovery.type": single-node
+    environment:
+      ELASTICSEARCH_GEM_VERSION: "<< parameters.elasticsearch_gem_version >>"
     steps:
       - checkout
-      - restore_cache: { keys: ['bundle-{{ checksum "Gemfile.lock" }}'] }
-      - run:
-          name: Wait for ES to be ready
-          command: |
-            until curl $ELASTICSEARCH_URL/_cat/health | egrep '(green|yellow)' 2>&1 > /dev/null; do
-              echo -n .
-              sleep 1
-            done
-      - run: |
-          bundle exec rspec --pattern "**/*_spec.rb" --format "progress"
+      - run: bin/setup
+      - run: bin/rubocop
+      - run: dockerize -wait http://localhost:9200 -timeout 1m
+      - run: bin/rspec
 
 workflows:
   version: 2
-  build:
+  test:
     jobs:
-      - bundle
-      - rubocop:
-          requires: [bundle]
-      - tests:
-          requires: [bundle]
-      - tests_7:
-          requires: [bundle]
+      - test:
+          name: "test with elasticsearch v7
+            and gem << matrix.elasticsearch_gem_version >>
+            and ruby v<< matrix.ruby_version >>"
+          elasticsearch_version: "7.17.5"
+          matrix:
+            parameters:
+              ruby_version: ["3.1", "2.7"]
```

Comment:
🤔 Should we also add `3.0`? Even if we don't use it in carwow, it is a supported Ruby version

Reply:
Could do, but I feel like it isn't worth it. We already run 6 test jobs because of all the permutations, and ruby 3.0 doesn't really differ much between 2.7 and 3.1, so it if works on both, pretty sure it works on 3.0. We can always add a job later when someone raises an issue that is 3.0 related.